src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -52,9 +54,12 @@ export default function withAnnotations<P extends object>(
         }
 
         handleAnnotationCreate = (eventData: AnnotationActionEvent): void => {
-            const { annotation = null, error = null } = eventData;
+            const { annotation = null, error = null, meta = null } = eventData;
+
             const action = this.getAction(eventData);
-            this.setState({ ...this.state, annotation, action, error });
+            const isPending = action === Action.CREATE_START;
+
+            this.setState({ ...this.state, annotation, action, error, isPending, meta });
```

Comment:
This is essentially storing the `action` value in state twice, once as a string and once as a boolean. Can we simplify?